### PR TITLE
refactor: remove acorn fork

### LINKED ISSUE #11439: Remove local fork of acorn JS parser
We currently have a local fork of acorn in-tree, but it would be more elegant to simply install it via npm.

Apparently there is a local patch that upstream don't want but a suggested workaround exists #11435

### PR DESCRIPTION
**Pull request by TroyTae**
resolve #11439

### CODE REVIEW

**Comment by TroyTae** [comment]
@welcome thank you bot
I'll add it after complete work :)

**Comment by TroyTae** [comment]
@kripken I'm sorry for asking.
Is there need to add `npm install` process or definite about acorn?

https://github.com/emscripten-core/emscripten/blob/master/tools/source-maps/sourcemap2json.js
It seems `source-map` just use `require('source-map')`

I tested below list, but all of things are failed:
1. move from devDep to dep in package.json
2. use relative path `require('../node_modeuls/acorn')`
3. add symbolic link `#!/usr/bin/env node`

**Comment by sbc100** [comment]
It should certainly be a dependencies rather than a devDependency since all users need it.

**Comment by sbc100** [comment]
The `#!/usr/bin/env node` should not been needed (or at least not for this change since we run this explicitly under the configured version of node (it also shouldn't hurt though).

**Comment by TroyTae** [comment]
I try to set `env` option using `env_with_node_in_path()`.
But it also failed..
Is there some hint about this??

**Comment by sbc100** [comment]
It looks like the npm-install step is running after we build all the libraries

Can you try moving npm-install from line 100 of `.circleci/config.yml` up to line 64?

**Comment by TroyTae** [comment]
Jesus, it was very simple issue 😂 Thank you!!
(I changed order of `npm-install` in `build` and `build-fastcomp`)
I'll request review after writing error handling code :)

**Comment by TroyTae** [comment]
@sbc100 @kripken
I'm ready for review!!

**Comment by kripken** [comment]
Code looks good, thanks @TroyTae !

What does the error message look like now - maybe paste an example here? (or is it identical to before?)

**Comment by TroyTae** [comment]
@kripken Error message is same with before.
There is already unit test about error message in here:
https://github.com/emscripten-core/emscripten/blob/98691d70014d41fefdcdcc9e7e720a6281e250c9/tests/test_other.py#L8580-L8602
And I wrote error handling code like this:
image

**Comment by kripken** [comment]
Great, thanks! Looks perfect then :smile:

**Comment by kripken** [comment]
This is ready to merge, please just add yourself to AUTHORS first.

**Comment by TroyTae** [comment]
Done 🚀

File: tools/acorn-optimizer.js [hunk 1]
```diff
@@ -1299,7 +1298,25 @@ if (extraInfoStart > 0) {
...
-var ast = acorn.parse(input, { ecmaVersion: 2018, preserveParens: closureFriendly, onComment: closureFriendly ? sourceComments : undefined });
+var ast;
+try {
+  ast = acorn.parse(input, {
+    ecmaVersion: 2018,
+    preserveParens: closureFriendly,
+    onComment: closureFriendly ? sourceComments : undefined
+  });
+} catch (err) {
+  err.message += (function() {
```
Comment (RReverser) [inline tools/acorn-optimizer.js:1309]: Oh, I haven't seen this PR. Great job!

Nitpick: a function expression is unnecessary here, I'd just use block-scoped variables (or regular ones, if compatibility with old Node version is concern). But doesn't matter much :)

Comment (TroyTae) [reply tools/acorn-optimizer.js:1309]: Yeah I also like to use `let` or `const`.
But every variable declaration uses `var` in this repo.
So I follow it and use IIFE to void hoisting 😅

**Comment by RReverser** [comment]
There was an issue in this PR - package-lock.json marks Acorn as a dev-dependency, making Emscripten fail for end users, even though it's a regular dependency in package.json.

Fixed up in https://github.com/emscripten-core/emscripten/pull/11520.

**Pull request merged**